plugin/PDFView.qml

```diff
@@ -142,6 +208,28 @@ SilicaFlickable {
             _feedbackEffect = null
         }
     }
+    onGotoClicked: {
+        var pt = base.contentAt(page - 1, top, left,
+                                Theme.paddingLarge, Theme.paddingLarge)
+        _contentYAtLinkTarget = pt.y
+        _pageAtGotoLink = pdfCanvas.currentPage
+        _contentXAtGotoLink = contentX
+        _contentYAtGotoLink = contentY
+        scrollTo(pt, page)
+    }
+    onContentYChanged: {
```

Comment:
Generally I'd avoid binding to contentY unless it's really needed, will be executed quite many times. How about storing target page instead of y position, and then follow how far pdfCanvas.currentPage is?